### PR TITLE
Add recipe for copilot.el

### PR DESCRIPTION
**Pull request by jgarte**
### Brief summary of what the package does

 The package provides a Copilot plugin for Emacs. 

### Direct link to the package repository

https://github.com/your/awesome_package

### Your association with the package

I am an enthusiastic user.

### Relevant communications with the upstream package maintainer

See the issue opened here and the community discussion:

https://github.com/copilot-emacs/copilot.el/issues/120

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [] My elisp byte-compiles cleanly
- [] I've used `M-x checkdoc` to check the package's documentation strings
- [] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

### CODE REVIEW

File: recipes/copilot [hunk 1]
```diff
@@ -0,0 +1 @@
+(mint-mode :fetcher github :repo "https://github.com/copilot-emacs/copilot.el")
```
Comment (alexmurray) [inline recipes/copilot:1]: `s/mint-mode/copilot/` ?

Comment (jgarte) [reply recipes/copilot]: Oops, just force pushed. Thanks for catching that.

I copy pasted too quickly from an old MELPA submission of mine 🦆

**Comment by riscy** [comment]
I adjusted your recipe - please double-check for me.  Otherwise, quick first pass of the code itself -

⸺ `copilot.el` with package-lint 20240923.1044 and `package-lint-main-file` = "copilot.el":
```
2 issues found:
1:0: warning: Including "Emacs" in the package summary is usually redundant.
125:13: error: You should depend on (emacs "29.1") if you need `python-ts-mode'.
```

⸺ `copilot.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- copilot.el#L797: You may want to prefer `(derived-mode-p 'xyz)`
```

**Comment by riscy** [comment]
Friendly ping. :)

**Comment by jgarte** [comment]
Hi,

Feel free to update this branch before me if you have the time, otherwise I'll have to wait till I find time later this week/weekend.

**Comment by jgarte** [comment]
@riscy Hi, what is there left to do?

**Comment by riscy** [comment]
> Feel free to update this branch before me if you have the time

Hi @jgarte - I didn't think the above message was meant for me in case there's some confusion on my side.

⸺ `copilot.el` with package-lint 20241127.1826 and `package-lint-main-file` = "copilot.el":
```
2 issues found:
1:0: warning: Including "Emacs" in the package summary is usually redundant.
125:13: error: You should depend on (emacs "29.1") if you need `python-ts-mode'.
```
Easy fixes, both of these:
- Your summary line is "An unofficial Copilot plugin for Emacs" and package-lint is telling you that "for Emacs" is redundant; a surprisingly large number of packages will describe themselves as "x for Emacs" and we get a cleaner package index when these descriptions are streamlined
- I browsed your code and I think this "error" on `python-ts-mode` is a false positive

⸺ `copilot.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- copilot.el#L797: You may want to prefer `(derived-mode-p 'xyz)`
```
I suggest this one because descendants of vterm-mode - should they ever arise - will then also benefit from the special handling you've coded in, but I don't consider this a blocker (as you can see from the gentle phrasing).

**Comment by LaurenceWarne** [comment]
Hi @riscy I think these should be addressed now as of https://github.com/copilot-emacs/copilot.el/pull/350 :slightly_smiling_face: (as you mention above the error with `python-ts-mode` looks to be a false positive to me too)

**Comment by jgarte** [comment]
Hi all, closing this issue. Thanks for working on this in the separate ticket and getting it merged/done.

**Comment by LaurenceWarne** [comment]
@jgarte looks like you closed this by mistake?

**Comment by jgarte** [comment]
@LaurenceWarne Do you want to rebase this ticket

**Comment by riscy** [comment]
Thank you!

**Pull request merged**